spatial/r3/euler_example_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3_test
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/num/quat"
+	"gonum.org/v1/gonum/spatial/r3"
+)
+
+// euler returns an r3.Rotation that corresponds to the Euler
+// angles alpha, beta and gamma which are rotations around the x,
+// y and z axes respectively. The order of rotations is x, y, z;
+// there are many conventions for this ordering.
+func euler(alpha, beta, gamma float64) r3.Rotation {
+	// Note that this function can be algebraically simplified
+	// to reduce floating point operations, but is left in this
+	// form for clarity.
+	var rot1, rot2, rot3 quat.Number
+	rot1.Imag, rot1.Real = math.Sincos(alpha / 2) // x-axis rotation
+	rot2.Jmag, rot2.Real = math.Sincos(beta / 2)  // y-axis rotation
+	rot3.Kmag, rot3.Real = math.Sincos(gamma / 2) // z-axis rotation
+
+	return r3.Rotation(quat.Mul(rot3, quat.Mul(rot2, rot1))) // order of rotations
+}
+
+func ExampleRotation_eulerAngles() {
+	// It is possible to interconvert between the quaternion representation
+	// of a rotation and Euler angles, but this leads to a variety of problem.
+	// The first of these is that there are a variety of conventions for
```

Comment:
_is_ or _are_ (a variety of conventions)?

Reply:
This is correct (at least for my dialect). I did a quick check and it seems prevalent. I think it would be singular if the object were uncountable, though maybe in American English it's singular.